aioshelly/rpc_device.py

```diff
@@ -0,0 +1,202 @@
+"""Shelly Gen2 RPC based device."""
+from __future__ import annotations
+
+from typing import Any, Dict
+
+import aiohttp
+
+from .common import ConnectionOptions, get_info
+from .exceptions import AuthRequired, NotInitialized
+from .wsrpc import WsRPC
+
+
+def mergedicts(dict1, dict2):
+    """Deep dicts merge."""
+    for k in set(dict1.keys()).union(dict2.keys()):
+        if k in dict1 and k in dict2:
+            if isinstance(dict1[k], dict) and isinstance(dict2[k], dict):
+                yield (k, dict(mergedicts(dict1[k], dict2[k])))
```

Comment:
why is this yielding? Now you're always going to call it with `dict` wrapping the generator ?